sigstore/_internal/oidc/__init__.py

```diff
@@ -40,6 +46,7 @@ def __init__(self, identity_token: str) -> None:
         """
         identity_jwt = jwt.decode(identity_token, options={"verify_signature": False})
 
+        self.exp_timestamp = identity_jwt.get("exp")
```

Comment:
We also need to retrieve the `nbf` claim here and check it below 🙂 

(Maybe we can push all of this into the `jwt.decode(...)` call instead? It should allow us to verify claims even through we're not verifying the signature.)